### PR TITLE
helm: add custom volume mounts

### PR DESCRIPTION
**Pull request by nolotz**
Hi there,

Similar to how we had it in the old ingress charts, adds this pr custom volume mounts to the new gateway.

Without this change, secrets are difficult to add in Envoy Filters.

### CODE REVIEW

**Review by morvencao — COMMENTED** [review]
/ok-to-test

File: releasenotes/notes/39493.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: release-notes/v2
+
+kind: feature
+
+area: installation
+
+issue:
+  - 39493
```
Comment (GregHanson) [inline releasenotes/notes/39493.yaml:8]: this should be a number reference to an issue on https://github.com/istio/istio/issues, not the PR#.  If there is not an associated github issue, can be left blank

Comment (nolotz) [reply releasenotes/notes/39493.yaml:8]: Hi @GregHanson,
https://github.com/istio/istio/issues/39493 should work though. 

For GitHub, issues and pull requests are the same and having the pull request linked would add contextual value.

**Comment by zirain** [comment]
maybe we should follow this https://github.com/istio/istio/issues/39180#issuecomment-1142441351

**Comment by nolotz** [comment]
> maybe we should follow this [#39180 (comment)](https://github.com/istio/istio/issues/39180#issuecomment-1142441351)

Hi @zirain,

I previously saw comments from @howardjohn and was already praying.
I'm fine with that even though such things as extraVolumes are already kinda standards in helm files.

A polished & clean helm will protect you better in battle than a dirty one.

**Pull request closed without merging**